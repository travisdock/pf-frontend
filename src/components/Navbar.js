import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import MoreMenu from './MoreMenu';

class Navbar extends React.Component {
  
  render() {
    return (
      <div className="navbar">
        { !!this.props.current_user.user_id
          ?
          <div className="navlinks">
            <NavLink
              to='/dashboard/form'
            >Input</NavLink>
            <NavLink
              to='/dashboard/charts'
            >Charts</NavLink>
            <NavLink
              to='/dashboard/entries'
            >Entries</NavLink>
            <MoreMenu />
          </div>
          :
          <div className="navlinks">
            <NavLink
              to='/login'
            >Login</NavLink>
            <NavLink
              to='/about'
            >About</NavLink>
            <NavLink
              to='/signup'
            >Signup</NavLink>
          </div>
        }
        </div>
      )
  }
}

const mapStateToProps = state => ({
  current_user: state.current_user
});
export default withRouter(connect(mapStateToProps)(Navbar))