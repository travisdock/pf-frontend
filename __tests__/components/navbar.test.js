import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components
import { MemoryRouter, NavLink } from 'react-router';

import Navbar from '../../src/components/navbar';

// Setup Store
const mockStore = configureStore();
const initialState = {current_user: {}}
const store = mockStore(initialState);

// props for no redux
const userProp = {user_id: null}
const historyProp = {push: jest.fn()}

describe('Navbar', () => {
    it('renders the component', () => {
        const wrapper = mount(
            <MemoryRouter 
              // this fakes the key in history to be testKey every time so it doesn't mess
              // up the snapshot.
              // https://github.com/ReactTraining/react-router/issues/5579#issuecomment-333401692
              initialEntries={[ { pathname: '/', key: 'testKey' } ]}
            >
                <Navbar store={store} current_user={userProp} />
            </MemoryRouter>
        );


        expect(toJson(wrapper)).toMatchSnapshot();
        // expect(wrapper.find(NavLink).props().to).toBe('/login');
    });
});