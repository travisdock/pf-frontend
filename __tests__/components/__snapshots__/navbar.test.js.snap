// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the component 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Connect(Navbar))
      current_user={
        Object {
          "user_id": null,
        }
      }
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Route>
        <Connect(Navbar)
          current_user={
            Object {
              "user_id": null,
            }
          }
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <Navbar
            current_user={Object {}}
            dispatch={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/",
                    "search": "",
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            store={
              Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              }
            }
            storeSubscription={
              Subscription {
                "listeners": Object {
                  "clear": [Function],
                  "get": [Function],
                  "notify": [Function],
                  "subscribe": [Function],
                },
                "onStateChange": [Function],
                "parentSub": undefined,
                "store": Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                },
                "unsubscribe": [Function],
              }
            }
          >
            <div
              className="navbar"
            >
              <div
                className="navlinks"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/login"
                >
                  <Route
                    path="\\\\/login"
                  >
                    <Link
                      aria-current={null}
                      replace={false}
                      to="/login"
                    >
                      <a
                        aria-current={null}
                        href="/login"
                        onClick={[Function]}
                      >
                        Login
                      </a>
                    </Link>
                  </Route>
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/about"
                >
                  <Route
                    path="\\\\/about"
                  >
                    <Link
                      aria-current={null}
                      replace={false}
                      to="/about"
                    >
                      <a
                        aria-current={null}
                        href="/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                    </Link>
                  </Route>
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/signup"
                >
                  <Route
                    path="\\\\/signup"
                  >
                    <Link
                      aria-current={null}
                      replace={false}
                      to="/signup"
                    >
                      <a
                        aria-current={null}
                        href="/signup"
                        onClick={[Function]}
                      >
                        Signup
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </div>
            </div>
          </Navbar>
        </Connect(Navbar)>
      </Route>
    </withRouter(Connect(Navbar))>
  </Router>
</MemoryRouter>
`;
